MbDotNet/Models/Imposters/RetrievedImposter.cs

```diff
@@ -1,16 +1,20 @@
-﻿using MbDotNet.Models.Requests;
+﻿using System.Collections.Generic;
+using MbDotNet.Models.Requests;
 using MbDotNet.Models.Responses.Fields;
+using MbDotNet.Models.Stubs;
 using Newtonsoft.Json;
 
 namespace MbDotNet.Models.Imposters
 {
     /// <summary>
     /// The base class for a retrieved imposter.
     /// </summary>
-    /// <typeparam name="T">The request type this imposter contains</typeparam>
+    /// <typeparam name="TRequest">The request type this imposter contains</typeparam>
+    /// <typeparam name="TResponseFields">The response fields type this imposter contains</typeparam>
     /// <typeparam name="TDefaultResponse">The request type this imposter contains</typeparam>
-    public abstract class RetrievedImposter<T, TDefaultResponse>
-        where T : Request
+    public abstract class RetrievedImposter<TRequest, TResponseFields, TDefaultResponse>
```

Comment:
Is there ever a situation where `TResponseFields` will be different from `TDefaultResponse`? I think we can support this feature without adding an additional generic type parameter.